src/main/java/io/jenkins/plugins/synopsys/security/scan/extension/freestyle/SecurityScanFreestyle.java

```diff
@@ -0,0 +1,428 @@
+    }
+
+    @DataBoundSetter
+    public void setPolaris_project_name(String polaris_project_name) {
+        this.polaris_project_name = Util.fixEmptyAndTrim(polaris_project_name);
+    }
+
+    @DataBoundSetter
+    public void setPolaris_assessment_types(String polaris_assessment_types) {
+        this.polaris_assessment_types = Util.fixEmptyAndTrim(polaris_assessment_types);
+    }
+
+    @DataBoundSetter
+    public void setPolaris_triage(String polaris_triage) {
+        this.polaris_triage = Util.fixEmptyAndTrim(polaris_triage);
+    }
+
+    @DataBoundSetter
+    public void setPolaris_branch_name(String polaris_branch_name) {
+        this.polaris_branch_name = Util.fixEmptyAndTrim(polaris_branch_name);
+    }
+
+    @DataBoundSetter
+    public void setBitbucket_token(String bitbucket_token) {
+        this.bitbucket_token = bitbucket_token;
+    }
+
+    @DataBoundSetter
+    public void setSynopsys_bridge_download_url(String synopsys_bridge_download_url) {
+        this.synopsys_bridge_download_url = synopsys_bridge_download_url;
+    }
+
+    @DataBoundSetter
+    public void setSynopsys_bridge_download_version(String synopsys_bridge_download_version) {
+        this.synopsys_bridge_download_version = synopsys_bridge_download_version;
+    }
+
+    @DataBoundSetter
+    public void setSynopsys_bridge_install_directory(String synopsys_bridge_install_directory) {
+        this.synopsys_bridge_install_directory = synopsys_bridge_install_directory;
+    }
+
+    @DataBoundSetter
+    public void setInclude_diagnostics(Boolean include_diagnostics) {
+        this.include_diagnostics = include_diagnostics ? true : null;
+    }
+
+    @DataBoundSetter
+    public void setNetwork_airgap(Boolean network_airgap) {
+        this.network_airgap = network_airgap ? true : null;
+    }
+
+    private Map<String, Object> getParametersMap(FilePath workspace, TaskListener listener)
+            throws PluginExceptionHandler {
+        return ScanParametersFactory.preparePipelineParametersMap(
+                this, ScanParametersFactory.getGlobalConfigurationValues(workspace, listener), listener);
+    }
+
+    @Override
+    public void perform(
+            @NonNull Run<?, ?> run,
+            @NonNull FilePath workspace,
+            @NonNull EnvVars env,
+            @NonNull Launcher launcher,
+            @NonNull TaskListener listener) {
+
+        listener.getLogger()
+                .println(
+                        "**************************** START EXECUTION OF SYNOPSYS SECURITY SCAN ****************************");
+
+        try {
+            ScanParametersFactory.createPipelineCommand(run, listener, env, launcher, null, workspace)
+                    .initializeScanner(getParametersMap(workspace, listener));
+        } catch (Exception e) {
+            if (e instanceof PluginExceptionHandler) {
+                try {
+                    throw new PluginExceptionHandler("Workflow failed! " + e.getMessage());
```

Comment:
Why are we wrapping only throw statement in try block?

Reply:
Hey @kishorikumar   Actually on compile time,  the **perform** method of **SecurityScanFreestyle** class forced us to surround the throw statement  with try/catch block.  That's the reason we put the throw statement in try/catch block. Me and @zaman-akib  also removed the try/catch block and tried to make it work in other ways. But it didn't work.  So as of now, it is the only way.  If you have any thing on your mind regarding this, you can share it with us.

Reply:
@kishorikumar 
It looks like this now,

`
if (e instanceof PluginExceptionHandler) {
                throw new RuntimeException(new PluginExceptionHandler("Workflow failed! " + e.getMessage()));
            } else {
                throw new RuntimeException(new ScannerException(ExceptionMessages.scannerFailureMessage(e.getMessage())));
            }
`